### PR TITLE
[Feature Enhanced] Add HammingLoss

### PR DESCRIPTION
**Pull request by WrRan**
Inspired by [`hamming_loss`](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/metrics/classification.py#L1794) in `sklearn`, I add the `hamming_loss` to metrics in AllenNLP.

### CODE REVIEW

**Comment by WrRan** [comment]
Please have a look. :)

**Comment by matt-gardner** [comment]
We don't see a concrete use case for this loss, so we're going to close this PR.  If someone has a concrete example where they want to use this, let us know.
